### PR TITLE
Adding oidc token & parsed oidc user as action outputs

### PR DESCRIPTION
**Pull request by carmithersh**
- [x] All [tests](https://github.com/jfrog/setup-jfrog-cli#build-the-code) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

Resolves https://github.com/jfrog/setup-jfrog-cli/issues/128
Supercedes https://github.com/jfrog/setup-jfrog-cli/pull/148

Add the OIDC token and parsed OIDC user as outputs of the Setup JFrog CLI action, so they can be utilized in non-CLI actions within the workflow.

Example usage:
```yml
- name: Setup JFrog CLI
  id: setup-jfrog-cli
  uses: ./
  env:
    JF_URL: ${{ secrets.JFROG_PLATFORM_URL }}
  with:
    oidc-provider-name: ${{ env.OIDC_PROVIDER_NAME }}

- name: Authenticate Docker
  uses: docker/login-action@v3
  with:
    registry: ${{ secrets.JFROG_PLATFORM_URL }}/docker
    username: ${{ steps.setup-jfrog-cli.outputs.oidc-user }}
    password: ${{ steps.setup-jfrog-cli.outputs.oidc-token }} 
```

### CODE REVIEW

File: .github/workflows/oidc-test.yml [hunk 1]
```diff
@@ -73,6 +72,17 @@ jobs:
...
+      - name: Test outputs
+        shell: bash
+        run: |
+            echo "os is $RUNNER_OS"           
+            if [ -z "${{ steps.setup.outputs.jf_oidc_token }}" ]; then
+              exit 1
+            fi
+            if [ -z "${{ steps.setup.outputs.jf_oidc_user }}" ]; then
+              exit 1
+            fi
```
Comment (yahavi) [inline .github/workflows/oidc-test.yml:84]:
```suggestion
      - name: Test User Output
        shell: bash
        run: test -n "${{ steps.setup-jfrog-cli.outputs.jf-oidc-user }}"

      - name: Test Token Output
        shell: bash
        run: test -n "${{ steps.setup-jfrog-cli.outputs.jf-oidc-token }}"
```

File: action.yml [hunk 2]
```diff
@@ -15,7 +15,11 @@ inputs:
     oidc-audience:
         description: "By default, this is the URL of the GitHub repository owner, such as the organization that owns the repository."
         required: false
-
+outputs:
+    jf_oidc_token:
```
Comment (yahavi) [inline action.yml:19]:
```suggestion
    jf-oidc-token:
```

File: action.yml [hunk 3]
```diff
@@ -15,7 +15,11 @@ inputs:
...
+        description: "JFrog oidc token generated by the JFrog CLI when setting oidc-provider-name."
+    jf_oidc_user:
```
Comment (yahavi) [inline action.yml:21]:
```suggestion
    jf-oidc-user:
```

File: action.yml [hunk 4]
```diff
@@ -15,7 +15,11 @@ inputs:
     oidc-audience:
         description: "By default, this is the URL of the GitHub repository owner, such as the organization that owns the repository."
         required: false
-
+outputs:
+    jf_oidc_token:
+        description: "JFrog oidc token generated by the JFrog CLI when setting oidc-provider-name."
```
Comment (yahavi) [inline action.yml:20]:
```suggestion
        description: "JFrog OIDC token generated by the Setup JFrog CLI when setting oidc-provider-name."
```

File: action.yml [hunk 5]
```diff
@@ -15,7 +15,11 @@ inputs:
...
+    jf_oidc_user:
+        description: "JFrog oidc username from the oidc token generated by the JFrog CLI when setting oidc-provider-name."
```
Comment (yahavi) [inline action.yml:22]:
```suggestion
        description: "JFrog OIDC username from the oidc token generated by the Setup JFrog CLI when setting oidc-provider-name."
```

File: .github/workflows/oidc-test.yml (hunk 1 above)
Comment (carmithersh) [reply .github/workflows/oidc-test.yml]: that did not work so I reverted it to my working test

**Pull request merged**